File: Deepspace-Ruby/lib/Loot.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'LootToUI'

module Deepspace
class Loot
  def initialize (  nsu, nw, nsh , nh , nm, ef=nil, city=nil)
    @nSupplies=nsu;
    @nWeapons=nw;
    @nShields=nsh;
    @nHangars=nh;
    @nMedals=nm;
    if ef.nil?
      @efficient=false
    else
      @efficient=ef
    end
    if city.nil?
      @SpaceCity=false
    else
      @spaceCity=city
    end
  end
  
  def getUIversion
    LootToUI.new(self)
  end
  
  def efficient
    @efficient
  end
  
  public
  def nSupplies
    @nSupplies
  end
  
  public
  def nWeapons
    @nWeapons
  end
  
  public
  def nShields
    @nShields
  end
  
  public 
  def nHangars
    @nHangars
  end
  
  public 
  def nMedals
    @nMedals
  end
  
  def spaceCity
    @spaceCity
  end
  
  public 
  def to_s
    return "Loot: nMedals=#{@nMedals}, nSupplies=#{@nSupplies}, nWeapons=#{@nWeapons}, 
    nShields=#{@nShields}, nHangars=#{@nHangars}, getEfficient=#{@getEfficient}, spaceCity=#{@spaceCity}"
  end
end
end

File: Deepspace-Java/src/deepspace/Loot.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author inaki
 */
public class Loot {
    private int nSupplies;
    private int nWeapons;
    private int nShields;
    private int nHangars;
    private int nMedals;
    private boolean getEfficient;
    private boolean spaceCity;
    
    Loot (int nsu, int nw, int nsh, int nh, int nm, boolean ef, boolean city){
        this.nSupplies=nsu;
        this.nWeapons=nw;
        this.nShields=nsh;
        this.nHangars=nh;
        this.nMedals=nm;
        this.getEfficient=ef;
        this.spaceCity=city;
        
    }
    
    Loot(int nsu, int nw, int nsh, int nh, int nm){
        this.nSupplies=nsu;
        this.nWeapons=nw;
        this.nShields=nsh;
        this.nHangars=nh;
        this.nMedals=nm;
    }
    
    LootToUI getUIversion(){
        return new LootToUI(this);
    }
    
    public boolean getEfficient(){
        return this.getEfficient;
    }
    
    public int getNSupplies(){
        return this.nSupplies;
    }
    
    public int getNWeapons(){
        return this.nWeapons;
    }
    
    public int getNShields(){
        return this.nShields;
    }
    
    public int getNHangars(){
        return this.nHangars;
    }
    
    public int getNMedals(){
        return this.nMedals;
    }
    
    public boolean spaceCity(){
        return this.spaceCity;
    }
    
    public String toString(){
        return "nSuplies: " + this.nSupplies +
               "nWeapons: " + this.nWeapons +
               "nShields: " + this.nShields +
               "nHangars: " + this.nHangars +
               "nMedals: " + this.nMedals +
               "getEfficient: "+ this.getEfficient +
                "spaceCity: "+ this.spaceCity;
    }
}
